### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
and some other house keeping chores.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# Define a container to hold any optimiser specific parameters
+# it's not a strict requirement - just a convention
+struct Descent{T}
+  η::T
+end
+
+# Define an `apply` rule with which to update the current params
+# using the gradients
+function Optimisers.apply(o::Descent, st, m, m̄)
+  m .- o.η .* m̄, st
```
Comment (ToucheSir) [inline README.md:32]:
```suggestion
function Optimisers.apply(o::Descent, state, m, m̄)
    o.η .* m̄, state
```
Slightly more descriptive parameter name, and removal of double-`patch`ing.

**Review by ToucheSir — COMMENTED** [review]
Mostly LGTM, just some minor stuff.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+## Usage
+
+```julia
+
+using Flux, Metalhead, Optimisers
+
+o = Optimisers.ADAM() # define an ADAM optimiser with default settings
+st = Optimisers.state(o, m)  # initialize the optimiser before using it
+
+model = ResNet().layers # define a model to train on
```
[the remark this answers is not in the document]
Comment (DhairyaLGandhi) [reply README.md:51]: What we really need is FluxML/Functors.jl#1

[the remark this answers is not in the document]
Comment (DhairyaLGandhi) [reply README.md:51]: We can drop pulling out the field.

**Comment by DhairyaLGandhi** [comment]
We should add some starting docs, we have API docs already, but maybe we should pull some out of Flux already?

**Comment by ToucheSir** [comment]
I think that's better served by a follow-up PR, since getting the site up will allow us more time to work on other docs stuff.

Are the failures on nightly expected or do they need to be resolved here?

**Comment by DhairyaLGandhi** [comment]
Nightly seems to started failing recently but I think that's an upstream issue which would get resolved soon. I hope in the meanwhile that we test out against 1.7 which is much more pressing IMO.

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**